### ISSUE TITLE
No console output after restore on a different host

### ISSUE THREAD

**Issue by digitalix** [checkpoint-restore/criu#958]
Restoring a process on a different host results in no output on stdout and stderr. 

At first, I thought it's something specific to the app I'm testing but then I built a very simple app in go to test this theory and there is no output in the console as well.

The test app:
```
func main() {
	for{
		log.Println(time.Now())
		time.Sleep(time.Second * 5)
	}
}
```
dump.log
https://pastebin.com/NLwLp06S

restore.log
https://pastebin.com/eHf343tS

```
criu dump -t 22 --shell-job --file-locks -vvv -o dump.log
criu restore --shell-job --file-locks -vvvv -o restore.log
```

Dump and restore is successful, but there is no output from the app. I also tried to redirect the stdout and stderr to a file but the result is the same.

**Comment by adrianreber** [comment]
Did you try this example: https://www.criu.org/Simple_loop

**Comment by digitalix** [comment]
> Did you try this example: https://www.criu.org/Simple_loop

Yes, this one works. The code I posted is basically the same so... what's going on?

**Comment by digitalix** [comment]
After a few hours, I noticed something very odd; once I restore the process on a new host it runs as it should but the output is for some reason discarded and after a few minutes the app starts to print out the logs.

Restoring the same process on the old host works fine. 

@adrianreber Any idea what could be the cause for this?

**Comment by adrianreber** [comment]
Strange, no ideas. Maybe it is related to the Go runtime. What does `lsof` say about the restored process and what do you see if you attach with `strace` to the restored process? Does it actually access stdout?

**Comment by digitalix** [comment]
@adrianreber 

I'm sure it's not related to the go runtime, the example in go was only to determine whether it's a problem with criu or ffmpeg which is what I'm testing.

Here is output from lsof:
https://pastebin.com/G3nLBJ4y

Looks like the files for stdout and stderr are open.

and strace
https://pastebin.com/uKW5hKg0

**Comment by avagin** [comment]
I think this can be related to monotonic or boot clocks. This will be fixed by time namespaces which have been merged to v5.6 kernel.

**Comment by digitalix** [comment]
I managed to solve the problem with ffmpeg by making a small change to the source code which was related to time, and now it seems to work. However, still not sure why a similar thing would happen too my go example.
